File: src/14.py
class Solution(object):
    def longestCommonPrefix(self, strs):
        """
        :type strs: List[str]
        :rtype: str
        """
        if len(strs) == 0:
            return ''
        if len(strs) == 1:
            return strs[0]
        
        prefix = ''
        
        for j in range(len(strs[0])):
            temp = strs[0][j]
            for i in range(len(strs)):
                if j >= len(strs[i]) or strs[i][j] != temp:
                    #注意or和and前后的条件顺序，会导致越界的事情
                    return prefix
                
            prefix += temp 
        return prefix

File: src/14.cpp
class Solution {
public:
    string longestCommonPrefix(vector<string>& strs) {
        if (strs.size() == 0) {
            return "";
        }
        if (strs.size() == 0) {
            return strs[0];
        }
        
        char temp;
        string prefix;
        
        for (int j = 0; j < strs[0].size(); j ++) {
            temp = strs[0][j];
            for (int i = 0; i < strs.size(); i ++) {
                if (j >= strs[i].size() || temp != strs[i][j]) {
                    return prefix;
                }
            }
            prefix.push_back(temp);
        }
        return prefix;
    }
};